collect_app/src/main/java/org/odk/collect/android/instances/Instance.java

```diff
@@ -91,7 +91,7 @@ public Builder canEditWhenComplete(boolean canEditWhenComplete) {
         }
 
         public Builder instanceFilePath(String instanceFilePath) {
-            this.instanceFilePath = new StoragePathProvider().getInstanceDbPath(instanceFilePath);
+            this.instanceFilePath = instanceFilePath;
```

Comment:
This scares me a little bit. It's not in use by Collect code other than in tests and in the method that builds an `Instance` from a DB record which means the path is already sure to be the correct one. So I think it's safe. But we have to be really careful if we do use it that we pass in the correct path.

This mirrors what `Form` does. I suspect that this had to be changed by @seadowg to test the `FormsRepository`. Without this change, I always got a crash from this method call when trying to test `InstancesRepository`.